src/lib.rs

```diff
@@ -1572,6 +1607,7 @@ impl<R: Read + Seek> MatroskaFile<R> {
             frame.is_discardable = queued_frame.is_discardable;
             frame.is_invisible = queued_frame.is_invisible;
             frame.is_keyframe = queued_frame.is_keyframe;
+            frame.duration = None; // Clean duration, as it's reported in different block than other frame content
```

Comment:
Is this comment really needed?

Reply:
I guess it's not essential, but since the duration is special (since it's in a separate block), I thought it could help reading the code.